### PR TITLE
Fish 7689 upgrading config 3.1 for mp 6.1

### PR DESCRIPTION
**Pull request by breakponchito**
## Description

This is to upgrade MP Config 3.1 and pass TCK
### Testing Performed

execution of payara-samples and MP TCK 3.1
unit tests for config from payara-samples:
`
Aug 25, 2023 10:23:43 AM org.jboss.arquillian.container.impl.MapObject populate
WARNING: Configuration contain properties not supported by the backing object fish.payara.arquillian.container.payara.CommonPayaraConfiguration
Unused property entries: {allowConnectingToRunningServer=true}
Supported property names: [adminHttps, debug, ignoreCertificates, libraries, type, target, adminUser, authorisation, domain, adminPort, properties, adminHost, adminPassword]
[INFO] Running fish.payara.samples.microprofile.config.MicroProfileConfigIT

[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.55 s - in fish.payara.samples.microprofile.config.MicroProfileConfigIT
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
`

results from Config 3.1 TCK:
`
[INFO] Results:
[INFO]

[INFO] Tests run: 374, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO]
[INFO] --- maven-failsafe-plugin:3.0.0-M5:verify (verify) @ config-tck ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
`

### Testing Environment

Linux ubuntu 20.04, azul jdk 11.0.12, maven 3.8.6

### CODE REVIEW

**Comment by breakponchito** [comment]
still working on unit testing

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
Jenkins test please

File: nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/ConfigExpressionResolver.java [hunk 1]
```diff
@@ -121,10 +130,17 @@ private ConfigValueImpl getValue(String propertyName) {
         for (ConfigSource source : sources) {
             final String result = source.getValue(propertyName);
             if (result != null && !result.isEmpty()) {
+                String resolvedExpression = null;
+                try {
+                    resolvedExpression = resolveExpression(result);
+                } catch(NoSuchElementException noSuchElementException) {
+                    log.warn("We don't need to thrown the exception just wrap the null result " +
```
Comment (aubi) [inline nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/ConfigExpressionResolver.java:137]: The message sounds strange, explaining what we don't need to do...?
I'd suggest message like "Using null value in configuration, expression %s".

Comment (breakponchito) [reply nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/ConfigExpressionResolver.java]: already done, thanks

**Pull request merged**